### ISSUE TITLE
Model evaluation - part 3

### ISSUE THREAD

**Issue by swenkel** [jhfjhfj1/autokeras#176]
### Bug Description

The issue remains that the test accuracy is 1.0.

It seems like the following two lines in ```metrics.py```
```python
        prediction = list(map(lambda x: x.argmax(), prediction))
        target = list(map(lambda x: x.argmax(), target))
```
mess it up and therefore have to be removed. If they are removed, then prediction and target are passed to ```accuracy_score``` directly  which is then yields the same as if I use it manually.

*I have not submitted a pull request since I am not entirely sure if that will mess up some other part of autokeras. As of now I would say: No. I am not sure if ```def _test``` in ```utils.py``` is used ever (for something relevant). But you certainly know the source code much better*.

### Reproducing Steps
  * Step 1: ...
  * Step 2: ...

### Expected Behavior

Show the real accuracy.

### Setup Details
Include the details about the versions of:
 - OS type and version:
 - Python: 
 - autokeras: 0.2.12, master, develop
 - scikit-learn:
 - numpy:
 - keras:
 - scipy:
 - tensorflow:
 - pytorch:
